frontend/src/pages/associations/elections/ActiveList.tsx

```diff
@@ -0,0 +1,246 @@
+                className={"float-right mt-0"}
+            >
+                <i className={"fe fe-edit"}/> Modifier
+            </Link>
+        );
+    } else {
+        editButton = (<></>)
+    }
+    const startsAt = new Date(election.startsAt);
+    const endsAt = new Date(election.endsAt);
+    return (
+        <Card>
+            <Card.Body>
+                <Row>
+                    <Col xs={0}>
+                        <h4>{election.name}</h4>
+                    </Col>
+                    <Col>
+                        {editButton}
+                    </Col>
+                </Row>
+                <Row>
+                    <p>Cette élection, ouverte le {startsAt.toLocaleDateString()} à {startsAt.toLocaleTimeString()}, fermera le {endsAt.toLocaleDateString()} à {endsAt.toLocaleTimeString()}. Les résultats seront immédiatement publiés.</p>
+                    <p>Vous avez le droit à un maximum de {election.maxChoicesPerBallot} choix</p>
+                </Row>
+                <Row>
+                    <Vote election={election} association={association}/>
+                </Row>
+                <Row>
+                    <ListOfVotersButtonModal election={election}/>
+                </Row>
+            </Card.Body>
+        </Card>
+    )
+};
+
+const Vote = ({ election, association, ...props }) => {
+    const isRegisterd = election.isRegistered;
+    const hasVoted = election.hasVoted;
+    const disabled = hasVoted || !isRegisterd;
+    const maxChoicesPerBallot = parseInt(election.maxChoicesPerBallot);
+
+    const [selected, setSelected] = useState<number[]>([]);
+    const [showVoteConfirmation, setShowVoteConfirmation] = useState<boolean>(false);
+    const newToast = useContext(ToastContext);
+
+    const disabledVoteButton = disabled || selected.length===0;
+
+
+    const handleClick = (choice: Choice) => {
+
+        const index = selected.findIndex((e)=>e===choice.id);
+        let newSelected: number[];
+        newSelected = selected.slice();
+        if (index >= 0) {
+            newSelected.splice(index, 1)
+        } else {
+            newSelected.push(choice.id)
+        }
+        if (newSelected.length > maxChoicesPerBallot) {
+            newSelected.shift()
+        }
+        setSelected(newSelected)
+    };
+
+    const handleShowAlert = () => {
+        setShowVoteConfirmation(true)
+    };
+
+    const handleVote = () => {
+        setShowVoteConfirmation(false);
+        const vote: Ballot = {
+            election: election.id,
+            choices: selected,
+        };
+        api.elections
+            .vote(vote)
+            .then(res => window.location.reload())
+            .catch(err => {
+                alert(err.message);
+                window.location.reload();
+            });
+
+    };
+
+    const handleCloseAlert = () => {
+        setShowVoteConfirmation(false)
+    };
+
+    return (
+        <>
+            {election.choices.map(choice => (
```

Comment:
This should be embedded in a `<Form>` and used with `Formik`.

Reply:
(see on a comment below what Formik is)

Reply:
I didn't ;(